File: weektwo/repeat/Queue.java
package repeat;

public class Queue {
    private int[] data;
    private int front;
    private int rear;
    private int capacity;
    private final static int DEFAULT_SIZE = 10;

    Queue() {
        this(DEFAULT_SIZE);
    }

    Queue(int size) {
        data = new int[size];
        front = -1;
        rear = -1;
        capacity = size;
    }

    boolean isFull() {
        return front == 0 && rear == capacity - 1;
    }

    boolean isEmpty() {
        return front == -1;
    }

    void enQueue(int value) throws StackExceptions {

        if (isFull()) {
            throw new StackExceptions("Stack is full");
        }

        if (front == -1) {
            front = 0;
        }
        data[++rear] = value;
    }

    int deQueue() throws StackExceptions {
        if(isEmpty()){
            throw new StackExceptions("Stack is Empty");
        }
        int val = data[front];
        if(rear==front){
            front = -1;
            rear = -1;
        }else{
            front++;
        }
        return val;
    }

    void display() throws StackExceptions {
        if(isEmpty()){
            throw new StackExceptions("Stack is Empty");
        }
        for (int i = front; i <=rear; i++) {
            System.out.print(data[i]+" --> ");
        }
        System.out.println();
    }

}


class Main2{
    public static void main(String[] args) throws StackExceptions {
        Queue que = new Queue();
        que.enQueue(23);
        que.enQueue(3);
        que.enQueue(16);
        que.enQueue(91);
        que.display();
        que.enQueue(2);
        que.display();

    }
}

File: dsaJavascript/queue/queue.js
class Queue {
  constructor() {
    this.items = [];
  }

  enqueue(data) {
    this.items.push(data);
  }

  dequeue() {
    if (!this.isEmpty()) {
      return this.items.shift();
    }
    return null;
  }

  peek() {
    if (!this.isEmpty()) {
      return this.items[0];
    }
    return null;
  }

  size() {
    return this.items.length;
  }

  clear() {
    this.items = [];
  }

  isEmpty() {
    return this.items.length === 0;
  }
}

const queue = new Queue();

queue.enqueue(10);
queue.enqueue(20);
queue.enqueue(30);

console.log(queue.items);

console.log(queue.peek());
console.log(queue.dequeue());
console.log(queue.peek());
